eglib/src/gmisc-unity.c

```diff
@@ -0,0 +1,19 @@
+#ifdef PLATFORM_UNITY
+
+#include <stdlib.h>
+#include <glib.h>
+
+#include "Path-c-api.h"
+
+static const char *tmp_dir;
+
+const gchar *
+g_get_tmp_dir(void)
+{
+    if (tmp_dir == NULL)
+        tmp_dir = UnityPalGetTempPath();
```

Comment:
I assume our UnityPalGetTempPath allocates memory...wondering if this should be
char* path = UnityPalGetTempPath();
tmp_dir = strdup(path);
delete path

I'm worried about mono calling g_free on strings that we hand it from il2cpp's allocator (though maybe those worries are unfounded?)